Auth0/NativeAuth.swift

```diff
@@ -22,20 +22,54 @@
 
 import UIKit
 
-public protocol AuthProvider {
-    func login(withConnection connection: String, scope: String, parameters: [String: Any]) -> NativeAuthTransaction
+/**
+ The NativeAuthCredentials struct defines the data requirements necessary to be returned
+ from a successul authentication with an IdP SDK as part of the NativeAuthTransaction process.
+ */
+public struct NativeAuthCredentials {
+    /// The OAuth2 access token returned by the IdP SDK
+    public let token: String
+    /// Any extras that may be necessary for certain Social IdP connections
+    public let extras: [String: Any]
 }
 
-public struct NativeAuthCredentials {
-    let token: String
-    let extras: [String: Any]
+/**
+ The NativeAuthTransaction expands on the AuthTransaction protocol to add additional functionality
+ required for handling Native Authentication scenarios.
+ 
+ The auth func handles the result from the IdP SDK authentication callback.
+ 
+ Scope, connnection and parameters are used in the final stage of Auth Transaction during the Auth0 Social Authentication.
+ 
+ ```
+ class SocialNativeTransaction: NativeAuthTransaction {
```

Comment:
I am not buying this example, it does not explain what each method does and what the parameter are for, what are the properties. Also it is not showing any real case usage.

We need to document each protocol property and method explaining what it does